ddtrace/internal/remoteconfig/client.py

```diff
@@ -227,19 +227,23 @@ def base64_to_struct(val, cls):
         self.converter.register_structure_hook(SignedRoot, base64_to_struct)
         self.converter.register_structure_hook(SignedTargets, base64_to_struct)
 
-        self._products = dict()  # type: MutableMapping[str, ProductCallback]
+        self._products = dict()  # type: MutableMapping[str, Optional[ProductCallback]]
         self._applied_configs = dict()  # type: Mapping[str, ConfigMetadata]
         self._last_targets_version = 0
         self._last_error = None  # type: Optional[str]
         self._backend_state = None  # type: Optional[str]
 
-    def register_product(self, product_name, func=None):
+    def register_product(self, product_name, func):
         # type: (str, Optional[ProductCallback]) -> None
         if func is not None:
             self._products[product_name] = func
         else:
             self._products.pop(product_name, None)
```

Comment:
```suggestion
        # type: (str, ProductCallback) -> None
        self._products[product_name] = func
```

is this what you intended?

I see we are adding `unregister_product`, which should cover if someone wants to explicitly remove it.

I also see that `self._products` allows `None` as the value, is there a scenario where we register a product but not a callback?

Reply:
Some mypy fight but I solved it and now it's like before.